### PR TITLE
Add basic JMH support

### PR DESCRIPTION
**Pull request by axel22**
This commit adds basic support for running the benchmarks from JMH.
I introduced a new project that depends on the main `renaissance` project, and uses it to list all the benchmarks and generate JMH wrapper classes. It then produces a fat JAR similar to the regular Renaissance JAR, but which includes the generated classes and the JMH's generated boilerplate code and resources.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -243,3 +249,134 @@ lazy val renaissance: Project = {
...
+      case PathList("scala", xs @ _*) =>
+        MergeStrategy.discard
+      case x =>
+        val oldStrategy = (assemblyMergeStrategy in assembly).value
+        oldStrategy(x)
+    }
+  )
+  .dependsOn(
+    renaissance
+  )
```
Comment (lbulej) [inline build.sbt]: Do you think that these chunks of JMH-specific code live in a separate `.scala` file (say `jmh.scala`) in the project directory?

Comment (axel22) [reply build.sbt]: That's a good idea - I will move them.

File: benchmarks/actors/src/main/scala/org/renaissance/actors/AkkaUct.scala [hunk 2]
```diff
@@ -38,6 +38,7 @@ class AkkaUct extends RenaissanceBenchmark {
   }
 
   protected override def runIteration(config: Config): Unit = {
+    //sys.error(" --> " + Predef.getClass.getClassLoader + " --> " + this.getClass.getClassLoader)
```
Comment (lbulej) [inline benchmarks/actors/src/main/scala/org/renaissance/actors/AkkaUct.scala]: Not needed anymore?

Comment (axel22) [reply benchmarks/actors/src/main/scala/org/renaissance/actors/AkkaUct.scala]: Good catch - removed!

File: build.sbt (hunk 1 above)
Comment (axel22) [reply build.sbt]: Moved JMH-specific functionality to `jmh.scala`.

**Comment by lbulej** [comment]
The wrapper classes will need to change when the benchmark API changes, so I would sleep better if you also added a test to Travis to test that the JMH version works :-) Other than that, LGTM.

**Comment by axel22** [comment]
Sounds good, let me see if my Travis-fu is adequate :)

**Comment by axel22** [comment]
Seems to work: https://travis-ci.org/renaissance-benchmarks/renaissance/jobs/545994131

File: project/build.scala [hunk 3]
```diff
@@ -110,4 +114,59 @@ class BenchmarkInfo(val benchClass: Class[_ <: RenaissanceBenchmark]) {
...
+
+          val isEligible =
+            !excludePattern.matcher(clazz.getPackage.getName).matches() &&
+              benchBase.isAssignableFrom(clazz)
+          if (isEligible) {
+            // Print info to see what benchmarks are picked up by the build.
+            val benchClass = clazz.asSubclass(benchBase)
+            val info = new BenchmarkInfo(benchClass)
+            if (logger.nonEmpty) logBenchmark(info, logger.get)
+
```
Comment (davleopo) [inline project/build.scala]: empty newline?

File: renaissance-core/src/main/java/org/renaissance/util/BenchmarkLoader.java [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    }
+
+    public RenaissanceBenchmark loadBenchmark(String name) {
+        try {
+            final Info bench = benchmarksByName.get(name);
+            final ClassLoader loader = ModuleLoader.getForGroup(bench.group);
+            final Class<?> benchClass = loader.loadClass(bench.className);
+            final Object result = benchClass.getDeclaredConstructor().newInstance();
+
+            // Make current thread as independent of the harness as possible
```
Comment (davleopo) [inline renaissance-core/src/main/java/org/renaissance/util/BenchmarkLoader.java]: missing "."

File: project/build.scala (hunk 3 above)
Comment (axel22) [reply project/build.scala]: Removed.

File: renaissance-core/src/main/java/org/renaissance/util/BenchmarkLoader.java (hunk 4 above)
Comment (axel22) [reply renaissance-core/src/main/java/org/renaissance/util/BenchmarkLoader.java]: Fixed.

**Pull request merged**